### PR TITLE
Test Parallel Add P0 test for Diffuse Probe Grid component

### PR DESCRIPTION
**Pull request by SWMasterson**
Added P0 test for Diffuse Probe Grid component

pyRunFile ../../Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DiffuseProbeGridAdded.py
Report:
[SUCCESS] Success: Diffuse Probe Grid Entity successfully created
[SUCCESS] Success: Entity has a Diffuse Probe Grid component
[SUCCESS] Success: UNDO Entity creation success
[SUCCESS] Success: REDO Entity creation success
[SUCCESS] Success: Diffuse Probe Grid component disabled
[SUCCESS] Success: Entity has a Axis Aligned Box Shape component
[SUCCESS] Success: Diffuse Probe Grid component enabled
[SUCCESS] Success: Diffuse Probe Grid component disabled
[SUCCESS] Success: Entity has a Box Shape component
[SUCCESS] Success: Diffuse Probe Grid component enabled
[SUCCESS] Success: Entered game mode
[SUCCESS] Success: Exited game mode
[SUCCESS] Success: Entity is hidden
[SUCCESS] Success: Entity is visible
[SUCCESS] Success: Entity deleted
[SUCCESS] Success: UNDO deletion success
[SUCCESS] Success: REDO deletion success
Test result:  SUCCESS

.\python\python.cmd -m pytest --build-directory build_development\bin\profile  .\AutomatedTesting\Gem\PythonTests\Atom\TestSuite_Main.py
=== test session starts ====
platform win32 -- Python 3.7.10, pytest-5.3.2, py-1.9.0, pluggy-0.13.1
rootdir: F:\git\o3de, inifile: pytest.ini
plugins: mock-2.0.0, timeout-1.3.4, ly-test-tools-1.0.0
collected 22 items

AutomatedTesting\Gem\PythonTests\Atom\TestSuite_Main.py ....................... [100%]

=== 23 passed in 64.22s (0:01:04) ===

Signed-off-by: Sean Masterson <semaster@amazon.com>

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DiffuseProbeGridAdded.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        "Diffuse Probe Grid Entity failed to be created")
+    diffuse_probe_grid_component = (
+        "Entity has a Diffuse Probe Grid component",
+        "Entity failed to find Diffuse Probe Grid component")
+    diffuse_probe_grid_disabled = (
+        "Diffuse Probe Grid component disabled",
+        "Diffuse Probe Grid component was not disabled")
+    box_shape_component = (
+        "Entity has a Box Shape component",
+        "Entity did not have a Box Shape component")
```
Comment (smurly) [inline AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DiffuseProbeGridAdded.py:27]: does this get used or just the for loop test_shape?

**Review by smurly — APPROVED** [review]
looks OK.
box_shape_component tuple might be unused.

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DiffuseProbeGridAdded.py (hunk 1 above)
Comment (SWMasterson) [reply AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DiffuseProbeGridAdded.py:27]: Ah you're right, that's not necessary anymore. I'll remove.

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DiffuseProbeGridAdded.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        TestHelper.init_idle()
+        TestHelper.open_level("", "Base")
+
+        # Test steps begin.
+        # 1. Create a Diffuse Probe Grid entity with no components.
+        diffuse_probe_grid_entity = EditorEntity.create_editor_entity(AtomComponentProperties.diffuse_probe_grid())
+        Report.critical_result(Tests.diffuse_probe_grid_creation, diffuse_probe_grid_entity.exists())
+
+        # 2. Add a Diffuse Probe Grid component to Diffuse Probe Grid entity.
+        diffuse_probe_grid_component = diffuse_probe_grid_entity.add_component(AtomComponentProperties.diffuse_probe_grid())
```
Comment (jromnoa) [inline AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DiffuseProbeGridAdded.py:102]: while we don't follow specific line lengths, this one is still a bit too long, pyCharm defaults to 120 characters but PEP-8 suggests 79 characters (this is due to different monitor resolutions and how much code they can fit on the screen at one time)
we can stick to 120 but that would still make this too long, recommend formatting like this to reduce line length:
```
diffuse_probe_grid_component = diffuse_probe_grid_entity.add_component(
AtomComponentProperties.diffuse_probe_grid())
```

**Review by jromnoa — CHANGES_REQUESTED** [review]

File: AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DiffuseProbeGridAdded.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    11) Delete Diffuse Probe Grid entity.
+    12) UNDO deletion.
+    13) REDO deletion.
+    14) Look for errors.
+
+    :return: None
+    """
+
+    import azlmbr.legacy.general as general
+    import azlmbr.render as render
```
[the remark this answers is not in the document]
Comment (SWMasterson) [reply AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DiffuseProbeGridAdded.py]: Removed

**Review by jromnoa — APPROVED** [review]

**Pull request merged**